File: Back/app/schemas/prices.py
"""식자재 가격 API 스키마 — 홈 화면 "실시간 최저가 추천"."""
from __future__ import annotations

from pydantic import BaseModel


class IngredientPriceResponse(BaseModel):
    item_name: str
    price: int
    unit: str
    direction: str  # UP | DOWN | SAME
    change_percent: float
    source: str = "KAMIS"

File: Front/src/api/endpoints/prices.ts
// 식자재 가격 API — 홈 화면 "실시간 최저가 추천" (KAMIS 연동).
import { api } from "../../lib/api";

export interface IngredientPrice {
  item_name: string;
  price: number;
  unit: string;
  direction: "UP" | "DOWN" | "SAME";
  change_percent: number;
  source: string;
}

export async function getIngredientPrices(): Promise<IngredientPrice[]> {
  return api.get("prices/ingredients").json<IngredientPrice[]>();
}
